src/slider-new/helpers/useMove.ts

```diff
@@ -0,0 +1,246 @@
+import { disableTextSelection, restoreTextSelection } from "./textSelection";
+import React, { HTMLAttributes, useMemo, useRef } from "react";
+import { useGlobalListeners } from "./useGlobalListeners";
+import { isNull } from "../../utils";
+
+export interface BaseMoveEvent {
+  pointerType: "mouse" | "pen" | "touch" | "keyboard";
+}
+
+export interface MoveStartEvent extends BaseMoveEvent {
+  type: "movestart";
+}
+
+export interface MoveMoveEvent extends BaseMoveEvent {
+  type: "move";
+  deltaX: number;
+  deltaY: number;
+}
+
+export interface MoveEndEvent extends BaseMoveEvent {
+  type: "moveend";
+}
+
+export type MoveEvent = MoveStartEvent | MoveMoveEvent | MoveEndEvent;
+
+export interface MoveProps {
+  onMoveStart?: (e: MoveStartEvent) => void;
+  onMove: (e: MoveMoveEvent) => void;
+  onMoveEnd?: (e: MoveEndEvent) => void;
+}
+
+export function useMove(props: MoveProps): HTMLAttributes<HTMLElement> {
+  const { onMoveStart, onMove, onMoveEnd } = props;
+
+  const state = useRef<{
+    didMove: boolean;
+    lastPosition: { pageX: number; pageY: number } | null;
+    id: number | null;
+  }>({ didMove: false, lastPosition: null, id: null });
```

Comment:
Extract this type out, just looks a bit cleaner.

```ts

interface useMoveStateRef {
  didMove: boolean;
  lastPosition: { pageX: number; pageY: number } | null;
  id: number | null;
}


  const state = useRef<useMoveStateRef>({
    didMove: false,
    lastPosition: null,
    id: null,
  });
```

Reply:
I believe these won't be necessary.
Reakit will expose useMove on the next release.
We can import them directly, untill then I think it can stay as it is 🤔